src/test/scala/viper/gobra/PluginAwareChopperTests.scala

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2025 ETH Zurich.
+
+package viper.gobra
+
+import org.scalatest.Inside
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import viper.gobra.util.PluginAwareChopper
+import viper.silver.ast
+import viper.silver.ast.utility.chopper.Penalty
+
+class PluginAwareChopperTests extends AnyFunSuite with Matchers with Inside {
+  test("Domains ending in WellFoundedOrders are kept in chopped programs") {
+    val intVarDecl = ast.LocalVarDecl("i", ast.Int)()
+    val function = ast.Function("functionA", Seq(intVarDecl), ast.Bool, Seq.empty, Seq.empty, None)()
```

Comment:
Small detail: I'd add a simple decreases measure, e.g. `decreases i` such that the axioms are necessary. I know this is not relevant for the current implementation but it would be cool if this test remains correct for a more advanced implementation that actually looks at decreases measures and decides, e.g., based on the expression's type what WellFoundedOrder domains & axioms are necessary